purchase_order_variant_mgmt/tests/test_purchase_order_variant_mgmt.py

```diff
@@ -92,10 +99,9 @@ def test_modify_variants(self):
             len(wizard.variant_line_ids.filtered('product_uom_qty')), 2,
             "There should be two fields with any quantity in the wizard."
         )
-        wizard.variant_line_ids.filtered(
-            lambda x: x.product_id == product1).product_uom_qty = 0
-        wizard.variant_line_ids.filtered(
-            lambda x: x.product_id == product2).product_uom_qty = 10
+
+        wizard.variant_line_ids[0].product_uom_qty = 0
+        wizard.variant_line_ids[1].product_uom_qty = 10
```

Comment:
Was this change really needed ? IMO looks safer to assert a quantity according to its product than its order in a recordset.

Reply:
I revert the cahnge.

Reply:
I did this change again. because now, the wizard line has not field product_id, so the filtered was raising an error.

Reply:
I have written about the same, but you are right. The product in current 2D matrix widget can be stored, so this is the less worst solution.

Reply:
@simahawk this is a flaw in current matrix widget: you can't store a full record in the matrix. Only the fields in row, column and value are stored.

Reply:
I'm afraid it doesn't serve. We were discussing in this PR: https://github.com/OCA/web/pull/1022, although it was more an internal chat between my colleague @Yajo and I.

Reply:
FTR reported here https://github.com/OCA/web/issues/1084